xds/internal/xdsclient/xdsresource/unmarshal_rds.go

```diff
@@ -367,10 +381,17 @@ func routesProtoToSlice(routes []*v3routepb.Route, csps map[string]clusterspecif
 					// resource will be NACKed." - RLS in xDS design
 					return nil, nil, fmt.Errorf("route %+v, action %+v, specifies a cluster specifier plugin %+v that is not in Route Configuration", r, a, a.ClusterSpecifierPlugin)
 				}
+				if csps[a.ClusterSpecifierPlugin] == nil {
+					// "If a plugin is not supported but has is_optional set,
+					// then we will ignore any routes that point to that plugin"
+					logger.Warningf("route %+v references optional cluster specifier plugin %v, the route will be ignored", r, a.ClusterSpecifierPlugin)
+					continue
+				}
 				cspNames[a.ClusterSpecifierPlugin] = true
 				route.ClusterSpecifierPlugin = a.ClusterSpecifierPlugin
 			default:
-				return nil, nil, fmt.Errorf("route %+v, has an unknown ClusterSpecifier: %+v", r, a)
+				logger.Warningf("route %+v references unknown ClusterSpecifier %+v, the route will be ignored", r, a)
```

Comment:
Do we have other warning logs when we process xDS data?  Is this going to turn into spam?  @menghanl @easwars

Reply:
I based this warning log off what was currently in the codebase with regards to ignoring routes. Let me know, I'll leave for now.

Reply:
I'm concerned this will become log spam.  Is there precedent for logging (at higher priority than INFO) when we get an xDS response we would eventually ACK?

Reply:
Switched to INFO, as per discussed in QAZSE.